### DISCUSSION TITLE
Build a discord bot for better communication

### DISCUSSION THREAD

**Discussion by jvmazagao** [chakra-ui/chakra-ui#2680] [Ideas]
# Feature request
Hello awesome team, I'm here to talk about a discord bot to announce everything, such
- Awesome Feature 2.0.0
- Changelog 
- Bugs

**Comment by with-heart** [comment]
I'm gonna convert this to a discussion so we can keep issues strictly for codebase-related stuff! But yes, this is definitely a discussion I'd like us to have.

**Comment by ljosberinn** [comment]
Functionality I'd like to see:
`!component button` - links all below individually

`!component button docs` - links to the docs for button
`!component button source` - links to the button package on github
`!component button storybook` - links to the button on the deployed storybook once we have it

**Reply by jvmazagao** [reply]
But, for initial functionalities, what we need to build?

I'm thinking of something simple in the beginning like, announcements, changelogs, and after going to something like scripting, and search.

**Reply by ljosberinn** [reply]
what do you understand under "scripting" and "search"?

re announcements & changelogs: the bot could listen to webhooks from github, just like the regular github integration in discord, and on a new release, fetch the changeset info or something similar, and more or less 1:1 post that into #announcements?

**Reply by jvmazagao** [reply]
Sorry for the delay, I'm having some work problems every day, but, I thinking about something like you post, 

`!component button props` and show every button, 
`!open issue bug` and communicate directly with the GitHub and open an issue,
`!list pending reviews` and return the link 

Something like that

**Reply by ljosberinn** [reply]
`!open` feels a bit odd, I'd reverse it to `!issue open <type>` with `type` `bug` simply linking `https://github.com/chakra-ui/chakra-ui/issues/new?template=bug_report.md`

another idea:
`!issue find <term>` - lists the top 10 results for issues coming up through github search for `term`

similarily for `!list`, I'd prefer `!reviews list <state>`

**Comment by with-heart** [comment]
I'd love to have a command that gently suggests that a user search for their issue on GitHub, maybe with a link to the issues page or something. There are times when I know a user is posting about an issue that already exists, and I'd like to kind of nudge our community in the direction of self-help and self-research rather than us having to track down issues all the time, if that makes sense

**Reply by with-heart** [reply]
Sometimes I feel like a librarian helping people find books they could easily find on the shelves themselves 😆

**Reply by jvmazagao** [reply]
We can create something like, `!issue find <term>` and, based on the term, I guess we can find something in the body of the GitHub API request.

**Comment by jvmazagao** [comment]
Can we create a way to talk directly with the maintainers for questions?
